### ISSUE TITLE
Where are different recommend methods?How can I use them?

### ISSUE THREAD

**Issue by linh47** [efpm04013/dnnrec_eswa#1]
In the 4th ipynb file named "Try different methods",there was a variable `MYMEDIALITEPATH='/home/kirana/MyMediaLite/bin'`,which was used in 
`Global Average
cmd=f'{MYMEDIALITEPATH}/rating_prediction --training-file={TRAINFILEPATH} --recommender=GlobalAverage \
--test-file={TESTFILEPATH} --prediction-file {DATAPATH}/prediction_globalavg.txt'` 
and 
`User Average
cmd=f'{MYMEDIALITEPATH}/rating_prediction --training-file={TRAINFILEPATH} --recommender=UserAverage \
--test-file={TESTFILEPATH} --prediction-file {DATAPATH}/prediction_useravg.txt'`
and 
`Item KNN Pearson
cmd=f'{MYMEDIALITEPATH}/rating_prediction --training-file={TRAINFILEPATH} --recommender=ItemKNN \
--test-file={TESTFILEPATH} --prediction-file {DATAPATH}/prediction_itemknn_pearson.txt \
--recommender-options k=1 --recommender-options num_iter=1'
cmd=f'{MYMEDIALITEPATH}/rating_prediction --training-file={TRAINFILEPATH} --recommender=ItemKNN \
--test-file={TESTFILEPATH} --prediction-file {DATAPATH}/prediction_itemknn_pearson.txt \
--recommender-options k=1 --recommender-options num_iter=1'`
and so on...

It looked like using different txt files and parameters to change different recommend algorithm.But I wonder where are these recommend algorithm and how to use them in code.
Thank you for your sharing and your excellent work.

**Comment by linh47** [comment]
Are there some recommended Python files for this algorithm in this folder “/home/kirana/MyMediaLite/bin/rating_prediction ”?
When I run this statement  `!command {cmd}`, is there a file executed in the shell, and which algorithm and which input and output data are specified by parameters?
I saw the output of  `!command {cmd}`
was 
```
=loading_time 0.05
memory 4
ratings range: [0.5, 4]
training data: 1481 users, 2002 items, 32036 ratings, sparsity 98.91951
test data:     992 users, 560 items, 3458 ratings, sparsity 99.37752
SVDPlusPlus num_factors=10 regularization=0.015 bias_reg=0.33 frequency_regularization=False learn_rate=0.001 bias_learn_rate=0.7 learn_rate_decay=1 num_iter=30  training_time 00:00:05.5479390 RMSE 0.8254516 MAE 0.6589829 CBD 0.2426074 new users: RMSE 0.98784 MAE 0.7973 CBD 0.25686 new items: RMSE 1.05327 MAE 0.8426 CBD 0.27741  new users and items: RMSE 1.28447 MAE 1.09997 CBD 0.3394 testing_time 00:00:00.0097460
prediction_time 00:00:00.0045370

memory 6
>
```
It looked like an execution result of Python file.
Am I right?

**Comment by linh47** [comment]
Because I can’t get the file "df_side_cold_runother_cf.pkl" after this file "4. Try different methods.ipynb" is run, I can’t continue the experiment later, I need your help, thank you.

**Comment by efpm04013** [comment]
Hi Linh,

To get the values for the Global Average, User Average, Item KNN Pearson
method ,... which are some of the baseline methods, the Mymedialite library
was used. MYMEDIALITEPATH refers to the path on the system where the
executable for mymedialite exists. Same for the TRAINFILEPATH, TESTFILEPATH
that refer to the location of the train and test file paths respectively.
DATAPATH is where the predictions are being stored

Mymedialite is just to get the base method estimators so that we can
compare dnnrec against them

Thanks
Kiran

**Comment by efpm04013** [comment]
yes, when we say !command the string that command contains is executed in
the shell

**Comment by efpm04013** [comment]
that pkl file is saved from an earlier ipynb file run. Ensure it is under
the right folder.

**Comment by linh47** [comment]
@efpm04013  Thanks for your reply.It helped me a lot.
I found Mymedialite in Github.But I still have a question.Could you tell me **how to run Mymedialite in Linux shell**?I downloaded both binary packages and source codes of Mymedialite.But I don't know how to use these resources.
I’m using google colab,it is a linux server that provides free GPU computing.I can run your jupter notebook files in google colab.
When I used !command {cmd} or !mycommand,it told me this
**/bin/bash: rating_prediction: command not found.**

mycommand is 
rating_prediction --training-file=inp_train_filmtrust.csv --recommender=GlobalAverage \
--test-file=inp_test_filmtrust.csv --prediction-file prediction_globalavg.txt

it can work when I  add "!" at the beginning of  other command.Such as !pip install xxxxx

I don't know how to build the Mymedialite  for Linux.Could you tell me how to solve this problem?
Thank you again~

**Comment by linh47** [comment]
I ran the command on the windows system and got the results. I think I have temporarily solved the problem at hand.

**Comment by Sangeethapsgtech** [comment]
> @efpm04013 Thanks for your reply.It helped me a lot.
> I found Mymedialite in Github.But I still have a question.Could you tell me **how to run Mymedialite in Linux shell**?I downloaded both binary packages and source codes of Mymedialite.But I don't know how to use these resources.
> I’m using google colab,it is a linux server that provides free GPU computing.I can run your jupter notebook files in google colab.
> When I used !command {cmd} or !mycommand,it told me this
> **/bin/bash: rating_prediction: command not found.**
> 
> mycommand is
> rating_prediction --training-file=inp_train_filmtrust.csv --recommender=GlobalAverage
> --test-file=inp_test_filmtrust.csv --prediction-file prediction_globalavg.txt
> 
> it can work when I add "!" at the beginning of other command.Such as !pip install xxxxx
> 
> I don't know how to build the Mymedialite for Linux.Could you tell me how to solve this problem?
> Thank you again~

Even i tried executing in colab but /bin/bash: gdrive/My: No such file or directory is the error i get all other paths are recognized and working perfectly except this file

**Comment by efpm04013** [comment]
Please ensure that you are meeting the software requirements.

The code needs Ubuntu to run as it has been tested on Ubuntu alone. In case
you want to run the code on Windows, it should be possible but will require
some minor changes.

**Comment by Kabir-Ehsan** [comment]
I am getting "command' is not recognized as an internal or external command,
operable program or batch file." this error. after executing !command{cmd} command.
@linh47 @efpm04013 @Sangeethapsgtech

**Comment by muhammadmehranbinazam** [comment]
HI
I hope this message finds you in good health.
Please help me to find these missing files. I was unable to find these files.
Thanks in Advance
![image](https://user-images.githubusercontent.com/48127979/132084251-5b5425f9-2e26-42f2-9e88-0a62547be2de.png)
![image](https://user-images.githubusercontent.com/48127979/132084272-4c0016bc-8dea-42fd-ac01-eee64d4c5f5f.png)

**Comment by Sangeethapsgtech** [comment]
> HI
> I hope this message finds you in good health.
> Please help me to find these missing files. I was unable to find these files.
> Thanks in Advance
> ![image](https://user-images.githubusercontent.com/48127979/132084251-5b5425f9-2e26-42f2-9e88-0a62547be2de.png)
> ![image](https://user-images.githubusercontent.com/48127979/132084272-4c0016bc-8dea-42fd-ac01-eee64d4c5f5f.png)

You will get these pickle files only when you complete the execution of previous files.

**Comment by muhammadmehranbinazam** [comment]
I am facing this error help me to resolve this.
I am using Ubuntu
![image](https://user-images.githubusercontent.com/48127979/132085745-ee07d290-409b-4d9d-925f-17baad35e8d2.png)

**Comment by negar3** [comment]
> I ran the command on the windows system and got the results. I think I have temporarily solved the problem at hand.

Could you explain how you ran the command on the windows system in Jupyter and got the results?
